### DISCUSSION TITLE
502 Bad Gateway when using vaultwarden with nginx proxy

### DISCUSSION THREAD

**Discussion by Braste** [dani-garcia/vaultwarden#4579] [Q&A] [answered]
Hello. 
Yesterday I installed vaultwarden on my debian 12 server.
I am using nginx 1.22.1 as webserver and proxy to other services on the same machine. Nginx is NOT installed with docker.
I set up vaultwarden with the following command

`docker run -d --name vaultwarden -v /opt/vaultwarden/vw-data/:/data/ --restart unless-stopped -p 11080:11080 vaultwarden/server:latest`

My nginx configuration is this
```
upstream vaultwarden-default {
  zone vaultwarden-default 64k;
  server 127.0.0.1:11080;
  keepalive 2;
}
map $http_upgrade $connection_upgrade {
    default upgrade;
    ''      "";
}
server {
    server_name vaultwarden.crm-braun.de;
    listen 443 ssl http2;
    listen [::]:443 ssl http2;
    ssl_certificate /etc/letsencrypt/live/crm-braun.de/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/crm-braun.de/privkey.pem;
    include /etc/letsencrypt/options-ssl-nginx.conf;
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
    add_header Strict-Transport-Security "max-age=31536000" always;
    ssl_trusted_certificate /etc/letsencrypt/live/crm-braun.de/chain.pem;
    ssl_stapling on;
    ssl_stapling_verify on;
    include snippets/block_agents.conf;

    location / {
      proxy_pass          http://vaultwarden-default;
      proxy_http_version 1.1;
      proxy_set_header    Upgrade           $http_upgrade;
      proxy_set_header    Connection        $connection_upgrade;
      proxy_set_header    Host              $host;
      proxy_set_header    X-Real-IP         $remote_addr;
      proxy_set_header    X-Forwarded-For   $proxy_add_x_forwarded_for;
      proxy_set_header    X-Forwarded-Proto $scheme;  
    }
}
server {
    listen 80;
    listen [::]:80;
    server_name vaultwarden.crm-braun.de;
    return 301 https://$host$request_uri;
}
```
When I try to open https://vaultwarden.crm-braun.de I get a 502. All other services are working fine.

**Comment by stefan0xC** [comment, marked as answer]
> `docker run -d --name vaultwarden -v /opt/vaultwarden/vw-data/:/data/ --restart unless-stopped -p 11080:11080 vaultwarden/server:latest`

You are [publishing the wrong port](https://docs.docker.com/network/#published-ports). You might want to change it to `-p 127.0.0.1:11080:80` (since `ROCKET_PORT` is set to 80 in the container image).
https://github.com/dani-garcia/vaultwarden/blob/753a9e0baee3ba3e7b4e05f18d3259e010b68d62/docker/Dockerfile.j2#L203-L205

**Reply by Braste** [reply]
Thank you, I am slightly ashamed that the solution was so simple.
